File: web_app/types/notification.ts
// Notification types
export type NotificationType =
    | 'offer_accepted'
    | 'task_started'
    | 'task_completed'
    | 'payment_released'
    | 'review_received'
    | 'dispute_raised'
    | 'revision_requested'
    | 'progress_update'
    | 'offer_declined'
    | 'task_cancelled'
    | 'offer_withdrawn';

export interface Notification {
    id: string;
    userId: string;
    type: NotificationType;
    title: string;
    message: string;
    data?: Record<string, any>; // Flexible data payload
    taskId?: string; // Legacy fields (can be deprecated if moved to data)
    offerId?: string;
    read: boolean;
    created_at: string; // Matches backend JSON tag
    actionUrl?: string;
}

File: auction_system/inspiration/lib/models/eCommerce/notification/notification.dart
class NotificationModel {
  final int id;
  final String title;
  final String message;
  final String type;
  final String url;
  final String createdAt;
  final bool isRead;

  NotificationModel({
    required this.id,
    required this.title,
    required this.message,
    required this.type,
    required this.url,
    required this.createdAt,
    required this.isRead,
  });
}
